spec/rubocop/cop/style/require_order_spec.rb

```diff
@@ -141,4 +141,24 @@
       RUBY
     end
   end
+
+  context 'when modifier conditional `if` is used between `require`' do
+    it 'registers no offense' do
+      expect_no_offenses(<<~RUBY)
+        require 'a'
+        require 'c' if foo
+        require 'b'
```

Comment:
The one-liner modifier `require`s form looks like it can be ordering to the following.

```ruby
        require 'a'
        require 'b'
        require 'c' if foo
```

So, I think this respects cop's intentions better.

Reply:
What should we do in the following cases?
Should it be only for modifier conditional?

```ruby
require 'c'
if foo
  require 'a' if foo
end
require 'b' if foo
```

Reply:
IMHO, semantically both could be treated if the branch body is just only `require`s.